tests/api.test.ts

```diff
@@ -277,16 +328,12 @@ describe('request', () => {
     const errorData = {
       error: 'Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.',
     };
-    fetch.mockRejectOnce(() => new Promise((_, reject) => setTimeout(() => reject(JSON.stringify(errorData)))), 100);
-    try {
-      await request();
-    } catch (error) {
-      expect(error.message).toBe(errorData.error);
-    }
+    fetch.mockRejectOnce(() => new Promise((_, reject) => setTimeout(() => reject(JSON.stringify(errorData)), 100)));
+    expect(request()).rejects.toThrowError(errorData.error);
   });
 
   it('resolves with function returning object body and init headers', async () => {
-    fetch.mockResponseOnce(() => Promise.resolve({ body: 'ok', init: { headers: { ding: 'dang' } } }), {
+    fetch.mockResponseOnce(() => Promise.resolve<MockResponse>({ body: 'ok', headers: { ding: 'dang' } }), {
```

Comment:
This is the only breaking change I would like to keep and maybe document.

This is because the original typings had:
```typescript
export interface MockParams {
  status?: number;
  statusText?: string;
  headers?: string[][] | { [key: string]: string }; // HeadersInit
  url?: string;
  /** Set >= 1 to have redirected return true. Only applicable to Node.js */
  counter?: number;
}

export interface MockResponseInit extends MockParams {
  body?: string;
  init?: MockParams;
}
```
I changed this to:
```typescript
export interface MockParams {
  status?: number;
  statusText?: string;
  headers?: [string, string][] | Record<string, string>; // HeadersInit
  url?: string;
  counter?: number;
}

export interface MockResponse extends MockParams {
  body?: string;
}
```
since otherwise you have duplicated properties, making it very fuzzy imho.

Reply:
:+1: I think I'm on board with this, but let me stew on it just a bit.

Reply:
Yep, I've convinced myself, the way it was before made no real sense.  We should indeed document this, but the best place is probably in the release notes.